src/docset.rs

```diff
@@ -49,6 +49,25 @@ pub trait DocSet: Send {
         doc
     }
 
+    /// Seeks to the target if possible and returns true if the target is in the DocSet.
+    ///
+    /// Implementations may choose to advance past the target if target does not exist.
+    ///
+    /// DocSets that already have an efficient `seek` method don't need to implement `seek_exact`.
+    /// All wapper DocSets should forward `seek_exact` to the underlying DocSet.
+    ///
+    /// ## API Behaviour
+    /// If `seek_exact` is returning true, a call to `doc()` has to return target.
+    /// If `seek_exact` is returning false, a call to `doc()` may return the previous doc,
```

Comment:
can you clarify that the result should still be a document in the posting list or TERMINATED, and that it could be something before or after the target.

Reply:
We also need to clarify what happens if the target is lower than the current doc.

Reply:
actually it seem like your current implementation might return `self.doc()` that is not in the docset at all.

Reply:
I added some comments to clarify the contract